content/docs/api-reference/dvcfilesystem.md

```diff
@@ -0,0 +1,126 @@
+Git commit (any [revision](https://git-scm.com/docs/revisions) such as a branch
+or a tag name, a commit hash, or an [experiment name]).
+
+[experiment name]: /doc/command-reference/exp/run#-n
+
+## Opening a file
+
+```py
+>>> with fs.open("model.pkl") as fobj:
+        model = pickle.load(fobj)
+```
+
+This is similar to `dvc.api.open()` which returns a file-like object. Note that,
+unlike `dvc.api.open()`, the `mode` defaults to binary mode, i.e. `"rb"`. You
+can also specify `encoding` argument in case of text mode (`"r"`).
+
+## Reading a file
+
+```py
+>>> contents = fs.cat_file("get-started/data.xml")
+```
+
+This is similar to `dvc.api.read()`, but it returns the contents of the file as
+bytes instead of a string.
+
+## Listing all DVC-tracked files recursively
+
+```py
+>>> fs.find("/", detail=False, dvc_only=True)
+[
+    '/data/data.xml',
+    '/data/features/test.pkl',
+    '/data/features/train.pkl',
+    '/data/prepared/test.tsv',
+    '/data/prepared/train.tsv',
+    '/evaluation/importance.png',
+    '/model.pkl'
+]
+```
+
+This is similar to `dvc ls --recursive --dvc-only` CLI command. Note that the
+`"/"` is considered as the root of the Git repo. You can specify sub-paths to
+only return entries in that directory. Similarly, there is `fs.ls()` that is
+non-recursive.
+
+## Listing all files (including Git-tracked)
+
+```py
+>>> fs.find("/", detail=False)
+[
+    ...
+    '/.gitignore',
+    '/README.md',
+    '/data/.gitignore',
+    '/data/data.xml',
+    '/data/features/test.pkl',
+    '/data/features/train.pkl',
+    '/data/prepared/test.tsv',
+    '/data/prepared/train.tsv',
+    ...
+    '/evaluation/.gitignore',
+    '/evaluation/importance.png',
+    '/evaluation/plots/confusion_matrix.json',
+    '/evaluation/plots/precision_recall.json',
+    '/evaluation/plots/roc.json',
+    '/model.pkl',
+    ...
+]
+```
+
+This is similar to `dvc ls --recursive` CLI command. It returns all of the files
+tracked by DVC and Git and if filesystem is opened locally, it also includes the
+local untracked files.
+
+## Downloading a file or a directory
+
+```py
+>>> fs.get_file("data/data.xml", "data.xml")
+```
+
+This downloads "data/data.xml" file to the current working directory as
+"data.xml" file. The DVC-tracked files may be downloaded from the cache if it
+exists or may get streamed from the remote.
+
+```py
+>>> fs.get("data", "data", recursive=True)
+```
+
+This downloads all the files in "data" directory - be it Git-tracked or
+DVC-tracked into a local directory "data". Similarly, DVC might fetch files from
+remote if they don't exist in the cache.
+
+## API Reference
+
+As DVCFileSystem is based on [fsspec](https://filesystem-spec.readthedocs.io/),
+it is compatible with most of the APIs that it offers. Please check the fsspec's
+[API Reference](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem)
+for more details.
```

Comment:
Last note: we could add the term "methods" or something here, to clarify we mean the reference of the elements of the class/object, not some larger API.